Bottom: a684aadb6a9dbee81eb3948c7e5a4a07754a110d
Top:    1064ee659e4f30c134bab2080242550e26dc7781
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-08 00:35:01 +0000

Refresh of fixup-code-to-determine

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 112ab4d..7f29072 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1407,7 +1407,7 @@ static int
 GhidDrawLayerGroup (int group, const BoxType * screen)
 {
   render_priv *priv = gport->render_priv;
-  int i, rv = 1;
+  int i;
   int layernum;
   int side;
   struct poly_info info;
@@ -1417,6 +1417,7 @@ GhidDrawLayerGroup (int group, const BoxType * screen)
   int first_run = 1;
   int top_group = GetLayerGroupNumberBySide (TOP_SIDE);
   int bottom_group = GetLayerGroupNumberBySide (BOTTOM_SIDE);
+  bool is_outline;
 
   if (!gui->set_layer (0, group, 0))
     return 0;
@@ -1426,9 +1427,8 @@ GhidDrawLayerGroup (int group, const BoxType * screen)
     layernum = layers[i];
     Layer = PCB->Data->Layer + layers[i];
 
-    if (strcmp (Layer->Name, "outline") == 0 ||
-        strcmp (Layer->Name, "route") == 0)
-      rv = 0;
+    is_outline = strcmp (Layer->Name, "outline") == 0 ||
+                 strcmp (Layer->Name, "route") == 0;
 
     if (layernum < max_copper_layer && Layer->On) {
 
@@ -1437,7 +1437,7 @@ GhidDrawLayerGroup (int group, const BoxType * screen)
 
       first_run = 0;
 
-      if (rv && !TEST_FLAG (THINDRAWFLAG, PCB)) {
+      if (!is_outline && !TEST_FLAG (THINDRAWFLAG, PCB)) {
         /* Mask out drilled holes on this layer */
         hidgl_flush_triangles (priv->hidgl);
         glPushAttrib (GL_COLOR_BUFFER_BIT);
@@ -1460,7 +1460,7 @@ GhidDrawLayerGroup (int group, const BoxType * screen)
         gui->end_layer ();
         gui->set_layer (0, group, 0);
 
-        if (rv && !TEST_FLAG (THINDRAWFLAG, PCB)) {
+        if (!is_outline && !TEST_FLAG (THINDRAWFLAG, PCB)) {
           hidgl_flush_triangles (priv->hidgl);
           glPushAttrib (GL_COLOR_BUFFER_BIT);
           glColorMask (0, 0, 0, 0);
@@ -1473,7 +1473,7 @@ GhidDrawLayerGroup (int group, const BoxType * screen)
       }
 
       /* Draw pins, vias and pads on this layer */
-      if (!global_view_2d && rv) {
+      if (!global_view_2d && !is_outline) {
         if (PCB->PinOn) r_search (PCB->Data->pin_tree, screen, NULL, pin_inlayer_callback, Layer);
         if (PCB->ViaOn) r_search (PCB->Data->via_tree, screen, NULL, via_inlayer_callback, Layer);
         if (PCB->PinOn && group == top_group)
